### PR TITLE
Added new default functions for `configgrpc` file

### PR DESCRIPTION
**Pull request by AkhigbeEromo**
Description:
Added newDefault methods for structs in configgrpc package

Link to tracking Issue:
Closes https://github.com/open-telemetry/opentelemetry-collector/issues/9654

Testing: Tests were added for the NewDefault functions

### CODE REVIEW

File: config/configgrpc/configgrpc.go [hunk 1]
```diff
@@ -150,6 +180,13 @@ type ServerConfig struct {
 	IncludeMetadata bool `mapstructure:"include_metadata"`
 }
 
+// NewDefaultServerConfig() creates a new ServerConfig with any default values set
+func NewDefaultServerConfig() *ServerConfig {
+	return &ServerConfig{
```
Comment (TylerHelmuth) [inline config/configgrpc/configgrpc.go:185]: This function will need to call `confignet.NewDefaultAddrConfig`

Comment (TylerHelmuth) [inline config/configgrpc/configgrpc.go:185]: This function will need to call `configauth.NewDefaultAuthentication` once it exists. We'll need to wait on this PR until that exists.

File: config/configgrpc/configgrpc.go [hunk 2]
```diff
@@ -46,6 +46,11 @@ type KeepaliveClientConfig struct {
 	PermitWithoutStream bool          `mapstructure:"permit_without_stream"`
 }
 
+// NewDefaultKeepaliveClientConfig() creates a new KeepaliveClientConfig with any default values set
+func NewDefaultKeepaliveClientConfig() *KeepaliveClientConfig {
+	return &KeepaliveClientConfig{}
```
Comment (TylerHelmuth) [inline config/configgrpc/configgrpc.go:51]: Based on https://pkg.go.dev/google.golang.org/grpc@v1.62.1/keepalive#ClientParameters I believe a default value of `10s` should be set.

File: config/configgrpc/configgrpc.go [hunk 3]
```diff
@@ -90,12 +95,27 @@ type ClientConfig struct {
 	Auth *configauth.Authentication `mapstructure:"auth"`
 }
 
+// NewDefaultClientConfig() creates a new ClientConfig with any default values set
+func NewDefaultClientConfig() *ClientConfig {
+	return &ClientConfig{
```
Comment (TylerHelmuth) [inline config/configgrpc/configgrpc.go:100]: This needs to use `configtls.NewDefaultClientConfig once it exists.

Comment (TylerHelmuth) [inline config/configgrpc/configgrpc.go:100]: This function will need to call configauth.NewDefaultAuthentication once it exists.

File: config/configgrpc/configgrpc.go [hunk 4]
```diff
@@ -107,6 +127,11 @@ type KeepaliveServerParameters struct {
 	Timeout               time.Duration `mapstructure:"timeout"`
 }
 
+// NewDefaultKeepaliveServerParameters() creates a new KeepaliveServerParameters with any default values set
+func NewDefaultKeepaliveServerParameters() *KeepaliveServerParameters {
+	return &KeepaliveServerParameters{}
```
Comment (TylerHelmuth) [inline config/configgrpc/configgrpc.go:132]: We should set some default values based on https://pkg.go.dev/google.golang.org/grpc@v1.62.1/keepalive#ServerParameters

File: config/configgrpc/configgrpc.go (hunk 1 above)
Comment (AkhigbeEromo) [reply config/configgrpc/configgrpc.go:185]: Understood, will add it immediately

Comment (AkhigbeEromo) [reply config/configgrpc/configgrpc.go:185]: Ohh, does an issue that will create `configauth.NewDefaultAuthentication` exist?

If it does can I get a link

Comment (TylerHelmuth) [reply config/configgrpc/configgrpc.go:185]: Ya it doesn't yet, we need to make it in `configauth` like you are for `configgrpc`.  See https://github.com/open-telemetry/opentelemetry-collector/issues/9821. You can work on that issue as well if you'd like.

File: config/configgrpc/configgrpc.go (hunk 3 above)
Comment (AkhigbeEromo) [reply config/configgrpc/configgrpc.go:100]: Concerning this too, can i get a link to the issue that will resolve this so I can work on it

File: config/configgrpc/configgrpc.go (hunk 2 above)
Comment (AkhigbeEromo) [reply config/configgrpc/configgrpc.go]: Done

File: config/configgrpc/configgrpc.go (hunk 1 above)
Comment (AkhigbeEromo) [reply config/configgrpc/configgrpc.go:195]: Just created a PR on this.
Here is the link: https://github.com/open-telemetry/opentelemetry-collector/pull/9825

File: config/configgrpc/configgrpc.go (hunk 3 above)
Comment (AkhigbeEromo) [reply config/configgrpc/configgrpc.go:104]: I have prepared a branch that will set default functions for `configtls`, but an issue has to first be created. Or should i go ahead and create the issue, then link a PR for the issue? @TylerHelmuth

**Comment by AkhigbeEromo** [comment]
Hello @TylerHelmuth, is there anything i need to do?

**Comment by TylerHelmuth** [comment]
@AkhigbeEromo We are waiting for https://github.com/open-telemetry/opentelemetry-collector/pull/9658 to be merged

**Comment by TylerHelmuth** [comment]
@AkhigbeEromo this is ready to use the `configtls. NewDefault*` funcs

**Comment by AkhigbeEromo** [comment]
I will have to rename this PR so I do not encounter the same issues @TylerHelmuth

**Pull request closed without merging**